src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -118,9 +118,9 @@ public void AddType(GraphType type, TypeCollectionContext context)
                 }
             });
 
-            if (type is ObjectGraphType)
+            var obj = type as ObjectGraphType;
+            if (obj != null)
```

Comment:
The pattern here was the direct casts so variables didn't leak into the outer scope  - as is such the case now with this change.